File: src/v4/Employee.java
package v4;

public class Employee extends Person {
    private double salary;
    private String employeeType;

    public Employee(String name, String nickname, String gender, String phoneNumber, String address, String email, double salary, String employeeType) {
        super(name, nickname, gender, phoneNumber, address, email);
        this.salary = salary;
        this.employeeType = employeeType;
    }

    // Getters and Setters for Employee properties
    public double getSalary() {
        return salary;
    }

    public void setSalary(double salary) {
        this.salary = salary;
    }

    public String getEmployeeType() {
        return employeeType;
    }

    public void setEmployeeType(String employeeType) {
        this.employeeType = employeeType;
    }

    @Override
    public String getGender() {
        return super.getGender();
    }

    @Override
    public void setGender(String gender) {
        super.setGender(gender);
    }

    @Override
    public int getId() {
        return super.getId();
    }

    @Override
    public String getAddress() {
        return super.getAddress();
    }

    @Override
    public String getEmail() {
        return super.getEmail();
    }

    @Override
    public String getName() {
        return super.getName();
    }

    @Override
    public String getNickname() {
        return super.getNickname();
    }

    @Override
    public String getPhoneNumber() {
        return super.getPhoneNumber();
    }

    @Override
    public void setAddress(String address) {
        super.setAddress(address);
    }

    @Override
    public void setEmail(String email) {
        super.setEmail(email);
    }

    @Override
    public void setName(String name) {
        super.setName(name);
    }

    @Override
    public void setNickname(String nickname) {
        super.setNickname(nickname);
    }

    @Override
    public void setPhoneNumber(String phoneNumber) {
        super.setPhoneNumber(phoneNumber);
    }
}

File: pythonCodeConversion/electricbill/models/employee.py
from models.person import Person


class Employee(Person):
    def __init__(self, name, nickname, gender, phone_number, address, email, salary, employee_type):
        super().__init__(name, nickname, gender, phone_number, address, email)
        self.salary = salary
        self.employeeType = employee_type

    # Getters and Setters for Employee properties
    def get_salary(self):
        return self.salary

    def set_salary(self, salary):
        self.salary = salary

    def get_employee_type(self):
        return self.employeeType

    # Override methods from the Person class
    def get_gender(self):
        return super().get_gender()

    def set_gender(self, gender):
        super().set_gender(gender)

    def get_id(self):
        return super().get_id()

    def get_address(self):
        return super().get_address()

    def get_email(self):
        return super().get_email()

    def get_name(self):
        return super().get_name()

    def get_nickname(self):
        return super().get_nickname()

    def get_phone_number(self):
        return super().get_phone_number()

    def set_address(self, address):
        super().set_address(address)

    def set_email(self, email):
        super().set_email(email)

    def set_name(self, name):
        super().set_name(name)

    def set_nickname(self, nickname):
        super().set_nickname(nickname)

    def set_phone_number(self, phone_number):
        super().set_phone_number(phone_number)
